Update REST API docs to explain why Wipe is free for Android (#50650)
fleetdm/fleet: docs/REST API/rest-api.md

Sends a command to wipe the specified macOS, iOS, iPadOS, Linux, Windows, or Android host. The host is wiped once it comes online.

> Wiping an company-owned (fully managed) Android host is available in Fleet Free and Fleet Premium; wipe is [free for company-owned Android and not other platforms](https://github.com/fleetdm/fleet/issues/41683) because "Wipe" used to be "Unenroll", which was free tier. 

Reviewer: I think let's simplify; feels a bit more more positive to simply say something is free rather than justify why it's free when nothing else is.
```suggestion
Wiping a company-owned (fully managed) Android host is available in Fleet Free and Fleet Premium. (This feature was [previously "Unenroll"](https://github.com/fleetdm/fleet/issues/41683).)
```

Author: Simplifying sounds good. I'm not sure about how much detail we want to go here for justification - we ran into a fair amount of confusion about why WIPE was free for Android BYOD while adding support for running/listing custom MDM commands for Android, and decided to give some context in the docs (linking to that Github issue) so we don't run into that again. WDYT @noahtalerman ?

Reviewer 2: > ran into a fair amount of confusion about why WIPE was free for Android BYOD while adding support for running/listing custom MDM commands for Android, and decided to give some context in the docs (linking to that Github issue) so we don't run into that again

Yup! I tweaked Rachael's edit by adding the link to Rachael's short language (I like it short).

What do you think?

Author: Looks good to me!
